src/convertToObject.js

```diff
@@ -6,7 +6,34 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const noEmptyLines = sourceString
+    .split(';')
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  // trim
+  const noSpaces = noEmptyLines.map((item) => {
+    const [property, value] = item.split(':');
+
+    const newKey = property.trim();
+    const newValue = value.trim();
+
+    return `${newKey}:${newValue}`;
+  });
+
+  const convertToObj = noSpaces.reduce((prev, item) => {
+    const [key, value] = item.split(':');
+
+    return { ...prev, [key]: value };
+  }, {});
```

Comment:
[CODE KNOWLEDGE] - Using the spread operator inside a `reduce` callback is inefficient because it creates a new copy of the object on each iteration. You should directly assign the new key-value pair to the `prev` (accumulator) object.